server.js

```diff
@@ -0,0 +1,178 @@
+		}))
+		.catch(error => response.status(500).json({
+			error: error.message
+		}))
+})
+
+app.get('/api/v1/stations/:station_id', (request, response) => {
+	const { station_id } = request.params
+
+	database('stations').where('station_id', station_id).select()
+		.then(stations => response.status(200).json(stations))
+		.catch(error => response.status(500).json({
+			error: error.message
+		}));
+})
+
+app.put('/api/v1/stations/:station_id', (request, response) => {
+	const newName = request.body.station_name;
+	const { station_id } = request.params
+	const station = database.where('station_id', station_id).select()
+	const oldName = station.station_name
+
+	if(!station) return response.status(404).json({ error: `station with id of ${station_id} was not found.`});
+	else if (!newName) return response.status(422).json({ error: 'No station name provided' });
+
+	database.where('station_name', oldName).update('station_name', newName)
+
+	response.status(202).json({
+		message: `Edit successful. Station with id of ${station_id} name changed from ${oldName} to ${newName}.`
+	})
+})
+
+app.delete('/api/v1/stations/:station_id', (request, response) => {
+	const { station_id } = request.params
+	database('stations').where('id', station_id).delete();
+		.then(station => response.status(200).json({
+			station,
+			message: `Station ${station_id} has been deleted.`
+		}))
+		.catch(error => response.status(500).json({
+			error: `Error deleting station: ${error.message}`
+		}))
+});
+
+// Cafe endpoints
+
+app.get('/api/v1/stations/:station_id/cafes', (request, response) => {
+	database('cafes').select()
+		.then(cafes => response.status(200).json(cafes))
+		.catch(error => response.status(500).json({
+			error: error.message
+		}));
+})
+
+app.post('/api/v1/stations/:station_id/cafes/:cafe_id', (request, response) => {
+	const cafe = request.body;
+
+	for(let requiredParam of [
+		'station_id',
+		'cafe_name',
+		'street_address', 
+		'city', 
+		'state', 
+		'zip_code',
+		'formatted_address', 
+		'cross_street', 
+		'distance_in_meters'
+		]) {
+		if(!cafe[requiredParam]) {
+			return response.status(422).json({
+				error: `Expected format: {
+					station_id: <Number>,
+					cafe_name: <String>,
+					street_address: <String>, 
+					city: <String>, 
+					state: <String>, 
+					zip_code: <String>,
+					formatted_address: <String>, 
+					cross_street: <String>, 
+					distance_in_meters: <String> }.  
+
+					You're missing the ${requiredParam} property.`
+			})
+		}
+	}
+
+	database('cafes').insert(cafe, 'id')
+		.then(cafeIds => response.status(201).json({
+			id: cafeIds[0],
+			message: `Project "${cafe.cafe_name}" successfully created!`
+		}))
+		.catch(error => response.status(500).json({
+			error: error.message
+		}))
+})
+
+
+
+app.get('/api/v1/stations/:station_id/cafes/:cafe_id', (request, response) => {
+	const { id } = request.params
+	database('cafes').where('id', id).select()
```

Comment:
Needs to be more specific in the `where` parens.
database('cafes').where({
    'id': cafe_id,
    'station_id': station_id
  }